runtime/src/test/scala/com/soundcloud/twinagle/ServerEndpointBuilderSpec.scala

```diff
@@ -0,0 +1,30 @@
+package com.soundcloud.twinagle
+
+import com.soundcloud.twinagle.test.TestMessage
+import com.twitter.finagle.Filter
+import com.twitter.finagle.http.{MediaType, Request, Status}
+import com.twitter.util.{Await, Future, Throw}
+import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
+
+class ServerEndpointBuilderSpec extends Specification {
+
+  "catches exceptions thrown by user code" in {
+    def oops(x: TestMessage): Future[TestMessage] = {
+      throw new Exception("oops")
+    }
+
+    val ex = new Exception("oh dear")
+    val builder = new ServerEndpointBuilder(_ => Filter.TypeAgnostic.Identity)
+    val (_, svc) = builder.build(
+      oops,
+      EndpointMetadata("/twirp", "service", "rpc")
+    )
+
+    val request = Request()
+    request.mediaType = MediaType.Json
+    request.contentString = "{}"
+
+    Await.result(svc(request).liftToTry) must not throwAn(ex)
```

Comment:
You could also just do `Await.ready` without the `liftToTry` (just for clarity: it took me a while to realize that this variant actually works as expected, too)

Reply:
`throwAn(ex)` but `ex` is never used anywhere? I'm not sure I get that part. I guess you can just remove the whole `must not throw` part, since throwing anything will automatically fail the test anyway

Reply:
Also, I'd probably use `throwA[Exception]`, it's not immediately clear that `throwAn(ex)` just compares the type, not the exception or message itself. Or just remove the whole `must not throw` part, since a throw will fail the test automatically anway.